### PR TITLE
implemented DynamicProxy as portrayed by the Scala reflection team

### PR DESCRIPTION
**Pull request by cvogt**
this requires Adriaan's applyDynamic pull request, so please merge together or this one after applyDynamic

### CODE REVIEW

File: src/library/scala/reflect/DynamicProxy.scala [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+ * A dynamic proxy which redirect method calls and attribute access to a given
+ * target object at runtime using reflection.
+ * A usage example can be found in test/files/run/dynamic-reflect.scala
+ * Not supported (yet):
+ *   - implicit conversions and parameters
+ *   - multiple arguments lists
+ *   - explicit type arguments
+ */
+trait DynamicProxy extends Dynamic{
+  val dynamicProxyTarget : AnyRef
```
Comment (dcsobral) [inline src/library/scala/reflect/DynamicProxy.scala]: This needs to be scaladoc'ed. It's the only member the user needs care about, and there's no info about it!

File: src/library/scala/reflect/DynamicProxy.scala [hunk 2]
```diff
@@ -0,0 +1,67 @@
+package scala.reflect
+/**
+ * A dynamic proxy which redirect method calls and attribute access to a given
+ * target object at runtime using reflection.
+ * A usage example can be found in test/files/run/dynamic-reflect.scala
```
Comment (dcsobral) [inline src/library/scala/reflect/DynamicProxy.scala]: Please, write usage examples in the scaladoc. Scala users need not download scala source files to learn how to use its features.

Comment (cvogt) [inline src/library/scala/reflect/DynamicProxy.scala]: thx for the comment. done.

File: src/library/scala/reflect/DynamicProxy.scala (hunk 1 above)
Comment (cvogt) [inline src/library/scala/reflect/DynamicProxy.scala]: like this :)? https://github.com/cvogt/scala/commit/66f75183c0614681c61eaa4f3af005be040ddd11#L0R18

**Pull request merged**